src/impl_aarch64_freebsd.c

```diff
@@ -0,0 +1,376 @@
+ID_AA64ISAR0_FEATURES(sha2_sha512, AARCH64_SHA2, AARCH64_SHA512)
+ID_AA64ISAR0_FEATURES(flagm, AARCH64_FLAGM)
+ID_AA64ISAR0_FEATURES(flagm2, AARCH64_FLAGM, AARCH64_FLAGM2)
+ID_AA64ISAR0_FEATURES(rng, AARCH64_RNG)
+
+static MrsFeature id_aa64isar0[] = {
+    ID_AA64ISAR0_SET_FEATURE(fhm, "FHM"),
+    ID_AA64ISAR0_SET_FEATURE(dp, "DP"),
+    ID_AA64ISAR0_SET_FEATURE(crc32, "CRC32"),
+    ID_AA64ISAR0_SET_FEATURE(rdm, "RDM"),
+    ID_AA64ISAR0_SET_FEATURE(atomics, "Atomic"),
+    ID_AA64ISAR0_SET_FEATURE(aes, "AES"),
+    ID_AA64ISAR0_SET_FEATURE(sha1, "SHA1"),
+    ID_AA64ISAR0_SET_FEATURE(sha2, "SHA2"),
+    ID_AA64ISAR0_SET_FEATURE(sha3, "SHA3"),
+    ID_AA64ISAR0_SET_FEATURE(sm3, "SM3"),
+    ID_AA64ISAR0_SET_FEATURE(sm4, "SM4"),
+    ID_AA64ISAR0_SET_FEATURE(aes_pmull, "AES+PMULL"),
+    ID_AA64ISAR0_SET_FEATURE(sha2_sha512, "SHA2+SHA512"),
+    ID_AA64ISAR0_SET_FEATURE(flagm, "CondM-8.4"),
+    ID_AA64ISAR0_SET_FEATURE(flagm2, "CondM-8.5"),
+    ID_AA64ISAR0_SET_FEATURE(rng, "RNG"),
+    MRS_FEATURE_END,
+};
+
+// ID_AA64ISAR1_EL1
+// -----------------------------------------------------------------------------
+ID_AA64ISAR1_FEATURES(i8mm, AARCH64_I8MM)
+ID_AA64ISAR1_FEATURES(dgh, AARCH64_DGH)
+ID_AA64ISAR1_FEATURES(bf16, AARCH64_BF16)
+ID_AA64ISAR1_FEATURES(sb, AARCH64_SB)
+ID_AA64ISAR1_FEATURES(frint, AARCH64_FRINT)
+ID_AA64ISAR1_FEATURES(pacg_gpi, AARCH64_PACG)
+ID_AA64ISAR1_FEATURES(pacg_gpa, AARCH64_PACG)
+ID_AA64ISAR1_FEATURES(api_pac, AARCH64_PACA)
+ID_AA64ISAR1_FEATURES(api_epac, AARCH64_PACA)
+ID_AA64ISAR1_FEATURES(api_epac2, AARCH64_PACA)
+ID_AA64ISAR1_FEATURES(api_fpac, AARCH64_PACA)
+ID_AA64ISAR1_FEATURES(api_fpac_combined, AARCH64_PACA)
+ID_AA64ISAR1_FEATURES(paca, AARCH64_PACA)
+ID_AA64ISAR1_FEATURES(lrcpc, AARCH64_LRCPC)
+ID_AA64ISAR1_FEATURES(ilrcpc, AARCH64_LRCPC, AARCH64_ILRCPC)
+ID_AA64ISAR1_FEATURES(fcma, AARCH64_FCMA)
+ID_AA64ISAR1_FEATURES(jscvt, AARCH64_JSCVT)
+ID_AA64ISAR1_FEATURES(dcpop, AARCH64_DCPOP)
+ID_AA64ISAR1_FEATURES(dcpodp, AARCH64_DCPOP, AARCH64_DCPODP)
+
+static MrsFeature id_aa64isar1[] = {
+    ID_AA64ISAR1_SET_FEATURE(i8mm, "I8MM"),
+    ID_AA64ISAR1_SET_FEATURE(dgh, "DGH"),
+    ID_AA64ISAR1_SET_FEATURE(bf16, "BF16"),
+    ID_AA64ISAR1_SET_FEATURE(sb, "SB"),
+    ID_AA64ISAR1_SET_FEATURE(frint, "FRINTTS"),
+    ID_AA64ISAR1_SET_FEATURE(pacg_gpi, "GPI"),
+    ID_AA64ISAR1_SET_FEATURE(pacg_gpa, "GPA"),
+    ID_AA64ISAR1_SET_FEATURE(lrcpc, "RCPC-8.3"),
+    ID_AA64ISAR1_SET_FEATURE(ilrcpc, "RCPC-8.4"),
+    ID_AA64ISAR1_SET_FEATURE(fcma, "FCMA"),
+    ID_AA64ISAR1_SET_FEATURE(jscvt, "JSCVT"),
+    ID_AA64ISAR1_SET_FEATURE(dcpop, "DCPoP"),
+    ID_AA64ISAR1_SET_FEATURE(dcpodp, "DCCVADP"),
+    ID_AA64ISAR1_SET_FEATURE(api_pac, "API PAC"),
+    ID_AA64ISAR1_SET_FEATURE(api_epac, "API EPAC"),
+    ID_AA64ISAR1_SET_FEATURE(api_epac2, "Impl PAuth+EPAC2"),
+    ID_AA64ISAR1_SET_FEATURE(api_fpac, "Impl PAuth+FPAC"),
+    ID_AA64ISAR1_SET_FEATURE(api_fpac_combined, "Impl PAuth+FPAC+Combined"),
```

Comment:
linux HWCAP idenifies only 0b0001 cases:

HWCAP_PACA
Functionality implied by ID_AA64ISAR1_EL1.APA == 0b0001 or ID_AA64ISAR1_EL1.API == 0b0001, as described by [Pointer authentication in AArch64 Linux](https://docs.kernel.org/arm64/pointer-authentication.html).

HWCAP_PACG
Functionality implied by ID_AA64ISAR1_EL1.GPA == 0b0001 or ID_AA64ISAR1_EL1.GPI == 0b0001, as described by [Pointer authentication in AArch64 Linux](https://docs.kernel.org/arm64/pointer-authentication.html).

we can add new fields and detecte it in Linux as well via check on HWCAP_CPUID, if this value is true then we can make access EL1 info in userland, for example, [pacg get info via inline asm](https://github.com/google/cpu_features/pull/186/files#diff-4344dc6b94f6ab65d13cc432f2151a88e6d14c9d44ced3a8e3ea53e107404ec9R159-R161)